ghost/member-attribution/lib/service.js

```diff
@@ -14,6 +14,42 @@ class MemberAttributionService {
         this.attributionBuilder = attributionBuilder;
     }
 
+    /**
+     *
+     * @param {Object} context instance of ghost framework context object
+     * @returns {Promise<import('./attribution').AttributionResource>}
+     */
+    async getAttributionFromContext(context) {
+        const source = this._resolveContextSource(context);
+
+        // We consider only select internal context sources
+        if (['import', 'api', 'admin'].includes(source)) {
+            let attribution = {};
+            if (source === 'import') {
+                attribution.referrerSource = 'Imported';
+            } else if (source === 'admin') {
+                attribution.referrerSource = 'Created Manually';
+            } else if (source === 'api') {
+                attribution.referrerSource = 'Created via API';
+            }
+
+            // If context has integration, set referrer medium as integration anme
+            if (context?.integration?.id) {
+                try {
+                    const integration = await this.models.Integration.findOne({id: context.integration.id}, {
+                        withRelated: ['api_keys', 'webhooks']
```

Comment:
Why are we fetching the api_keys and webhooks?

Reply:
Yea that'll go away in final cleanup, came in with when pasting for quick hack locally.

Reply:
Okay cool

Reply:
Updated ✅